### ISSUE TITLE
changedfiles: latest rpminspect is getting OOM-killed

### ISSUE THREAD

**Issue by msrb** [rpminspect/rpminspect#1197]
**Describe the bug**

It seems like 4e448f334b7217761f38a9e1676cc6fb3b6ba15e introduced some regression (in changedfiles inspection) that causes rpminspect to consume all available memory only to be OOM-killed immediately after that.

Versions before 4e448f334b7217761f38a9e1676cc6fb3b6ba15e don't exhibit this behavior.

The problem doesn't affect all tested components, but for example kernel is affected.

**To Reproduce**
```
$ /usr/bin/rpminspect -c /usr/share/rpminspect/redhat.yaml -T changedfiles --arch x86_64,noarch  --output=result.txt --verbose --release=el9 --profile=rhel-9 kernel-5.14.0-284.11.1.el9_2 52968362
...
=> kernel-devel-matched-5....4.11.1.el9_2.x86_64.rpm  [######################################################]
=> kernel-abi-stablelists-...4.11.1.el9_2.noarch.rpm  [######################################################]
=> kernel-doc-5.14.0-284.11.1.el9_2.noarch.rpm        [######################################################]
Skipping abidiff inspection...       skip
Skipping addedfiles inspection...    skip
Skipping annocheck inspection...     skip
Skipping arch inspection...          skip
Skipping badfuncs inspection...      skip
Skipping capabilities inspection...  skip
Killed

$ rpm -qa|grep rpminspect
rpminspect-data-centos-1.3-0.1.202303151248git53bfd31.fc38.noarch
rpminspect-data-fedora-1.11-0.1.202304251747gitcec0c97.fc38.noarch
rpminspect-data-redhat-1.8-0.1.202305250802gite1d03b4.fc38.noarch
librpminspect-1.12-0.1.202305311232git8c822b1.fc38.x86_64
rpminspect-1.12-0.1.202305311232git8c822b1.fc38.x86_64
```

My laptop has 32 gigs of memory.

**Expected behavior**
No OOM :)

**Environment:**
RHEL CI

**Comment by dcantrell** [comment]
Westford labs are still down, so I can't talk to Brew.  Have to wait for that to come back.  In the interim I am testing with some Fedora kernels.

**Comment by dcantrell** [comment]
Well, no.  Those are down too.  Fun.

**Comment by dcantrell** [comment]
This one is taking longer to fix than I wanted it to.  I currently have it passing now, but there is a leak I am trying to track down.

**Comment by dcantrell** [comment]
I've got one PR in progress and likely more to come to address this.  There were a number of problems and the first step was to get the leaks and SIGSEGV issues sorted.  Now I'm dealing with massive memory consumption.  I'm able to do a kernel build comparison locally (without restricting the architectures, just doing all of them) and the program gets up to consuming 108GB of memory.  This seems very wrong.

**Comment by dcantrell** [comment]
At one point I had rpminspect running for more than 72 hours to profile the memory usage.  It completed and it had recorded over 12 terabytes of allocations from rpminspect.  At this point I believe I have tracked down every error for this inspection, including the one mistake I made from the other commit, and kernel comparisons now are taking around 2 to 3 GB on my system.  This seems more reasonable.  Once the tests pass, I will merge this giant PR.
